packages/sdk/src/inscription/instant-buy.ts

```diff
@@ -227,27 +226,26 @@ export async function generateRefundableUTXOs({
 
   const utxo = spendableUTXOs.sort((a, b) => b.sats - a.sats)[0] // Largest UTXO
   const psbt = new bitcoin.Psbt({ network: networkObj })
-  const witnessScripts: Buffer[] = []
+  const witnesses: Buffer[] = []
   const input = await processInput({ utxo, pubKey: publicKey, network })
   const totalOutputs = 3
   const outputs: { address: string; cardinals: number }[] = []
 
-  input.witnessUtxo?.script && witnessScripts.push(input.witnessUtxo?.script)
+  input.type === "taproot" && input.witnesses && witnesses.push(...input.witnesses)
   psbt.addInput(input)
 
   if (enableRBF) {
     psbt.setInputSequence(0, 0xfffffffd) // hardcoded index because input is just one
   }
 
-  const fees = calculateTxFee({
-    totalInputs: 1,
-    totalOutputs,
-    satsPerByte: feeRate,
-    type: pubKeyType,
-    additional: { witnessScripts }
+  const feeEstimator = new FeeEstimator({
+    psbt,
+    feeRate,
+    network
   })
+  const fee = feeEstimator.calculateNetworkFee()
```

Comment:
can we also return the fee for refundable utxo?

Reply:
Done in #64